sleap/nn/tracking.py

```diff
@@ -1140,11 +1302,21 @@ def add_cli_parser_args(cls, parser, arg_scope: str = ""):
             else:
                 arg_name = arg["name"]
 
-            parser.add_argument(
-                f"--{arg_name}",
-                type=arg["type"],
-                help=help_string,
-            )
+            if arg["name"] == "tracker":
+                # If default is defined for "tracking.tracker", we cannot detect
+                # mal-formed command line.
+                parser.add_argument(
+                    f"--{arg_name}",
+                    type=arg["type"],
+                    help=help_string,
+                )
+            else:
+                parser.add_argument(
+                    f"--{arg_name}",
+                    type=arg["type"],
+                    help=help_string,
+                    default=arg["default"],
+                )
```

Comment:
Hi @getzze,

Could you please let us know why we need this catch?

Reply:
So, before, `add_argument` was called without the `default` argument, meaning that the default arguments defined in the `.yaml` file were not passed to the parser. Which is a bug I think. This PR adds specifying the `default` argument.

For the "tracking.tracker" option, default is set to the string "none". Before this PR, as the `default` argument was not used, the default was `None`.
"tracking.tracker" defaulting to `None` is used at least  once to detect ill-formed command line, [here](https://github.com/talmolab/sleap/blob/83d6bc04ce3f0c93a8af9a7536db39c60affc1d8/sleap/nn/inference.py#L5646-L5647) and [here](https://github.com/talmolab/sleap/blob/83d6bc04ce3f0c93a8af9a7536db39c60affc1d8/sleap/nn/inference.py#L5700-L5706). So I chose to keep the old `None` default just for this option, to be on the safe side :sweat_smile: 


```
    # running tracking on existing prediction file
    elif getattr(args, "tracking.tracker") is not None:
```

Reply:
great, thanks!